xdsl/transforms/canonicalization_patterns/arith.py

```diff
@@ -16,3 +17,28 @@ def match_and_rewrite(self, op: arith.Addi, rewriter: PatternRewriter) -> None:
             and value.value.data == 0
         ):
             rewriter.replace_matched_op([], [op.rhs])
+
+
+class FoldConstConstOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: arith.FloatingPointLikeBinaryOperation, rewriter: PatternRewriter, /
+    ):
+        if (
+            isinstance(op.lhs.owner, arith.Constant)
+            and isinstance(op.rhs.owner, arith.Constant)
+            and isa(l := op.lhs.owner.value, builtin.AnyFloatAttr)
```

Comment:
Might be easier to use `isinstance(l, FloatAttr)` here but doesn't make a lot of difference.

Reply:
Unfortunately, this will cause pyright to say that the type is partially unknown when using `l`, though I agree that isinstance is more lightweight and preferable.